setup.sh

```diff
@@ -1,56 +1,64 @@
 #!/bin/bash
-time=`date '+%Y%m%d_%H%M'`
-VAG_COMMON_DIR="/vagrant/"
+time=`TZ=Asia/Jerusalem date '+%Y%m%d_%H%M'`
+VAG_COMMON_DIR="/vagrant"
+HOME="/home/vagrant"
 LOG_DIR="$VAG_COMMON_DIR/log"
 OUTFILE="$LOG_DIR/${time}_vagrant-out.log"
+FLASK_APP_PY="$VAG_COMMON_DIR/flask_init.py"
+FLASK_DIR="$HOME/FlaskApp"
+FLASK_EXEC="$HOME/runFlask.sh"
+KILLFLASK_SCRIPT="$VAG_COMMON_DIR/killFlask.sh"
+PYTHON3="`which python3`"
+FLASK_ONBOOT=false # Currently Flask On Startup feature disabled [ Not Supported ]
 
 runSetup() {
     step=1
 
+    if [ ! -e $FLASK_APP_PY ]; then
+        echo "ERROR: Missing $FLASK_APP_PY file..." >> $OUTFILE
+        exit 1
+    else
+        chmod +x $FLASK_APP_PY
+    fi
+
+    if [ ! -d $FLASK_DIR ]; then
+        mkdir $FLASK_DIR
+    fi
+
     if [ ! -d $LOG_DIR ]; then
         mkdir $LOG_DIR
+    else
+        rm -f $LOG_DIR/*vagrant-out.log
     fi
-
-    if [ ! -e $OUTFILE ]; then
-        touch $OUTFILE
+    touch $OUTFILE
+    # Adding execution permission to $KILLFLASK_SCRIPT 
+    if [ -e $KILLFLASK_SCRIPT ]; then
+        chmod +x $KILLFLASK_SCRIPT
     fi
 
     echo " "
     echo " "
     echo "==> Running Provision Script: $0:"
     echo " "
-    echo "$((step++)). Updating apt-git repositories..."
-    sudo apt-get update  >> $OUTFILE 2>&1 
-    [ $? -ne 0 ] && exit 1
-    sudo apt-get -yq install software-properties-common >> $OUTFILE 2>&1 
-    [ $? -ne 0 ] && exit 1
+    echo "$((step++)). Installing & Activating Python-3 Virtual Environment..."
+    sudo apt-get install -yq python3-venv >> $OUTFILE 2>&1 
+    cd $FLASK_DIR
+    $PYTHON3 -m venv venv
```

Comment:
why are you using a virtualenv?

Reply:
As I replied to Liora:

In short, a best practice to isolate a Python project with its own required libraries to ensure all dependencies.
It's reasonable that this won't necessarily be an issue in our case, I just applied it per Flask documentation recommendation.

Taken from Flask Documentation
https://flask.palletsprojects.com/en/1.1.x/installation/

"The more Python projects you have, the more likely it is that you need to work with different versions of Python libraries, or even Python itself. Newer versions of libraries for one project can break compatibility in another project.

Virtual environments are independent groups of Python libraries, one for each project. Packages installed for one project will not affect other projects or the operating system’s packages."

Reply:
In general, its a very good idea. However, its not necessary when using vagrant if you are planning on using a single virtual machine per python project. It doesn't do any real harm, though, aside from adding complexity.

Reply:
Let me know should you prefer I remove it and perform tests accordingly.

Reply:
either way works. Personally I wouldn't, but if the group wants to keep it as a standard best practice, I'd let it in

Reply:
I believe the team will appreciate simplicity, removed it as of next commit.